src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,121 @@
 package core.basesyntax;
 
-/**
- * Feel free to remove this class and create your own.
- */
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import java.util.Random;
+
 public class HelloWorldTest {
+    private Calculator calculator;
+
+    @Before
+    public void setUp() {
+        calculator = new Calculator();
+    }
+
+    @Test(expected = ArithmeticException.class)
+    public void arithmeticException() {
+        try {
+            calculator.calculate(5, 0, '/');
+        } catch (ArithmeticException e) {
+            throw new ArithmeticException();
+        }
+        Assert.fail("Arithmetic Exception vas expected");
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void unexpectedOperation() {
+        try {
+            for (int i = 0; i < 128; i++) {
+                char operation = (char) i;
+                if (!(operation == '+'
+                        || operation == '-'
+                        || operation == '*'
+                        || operation == '/'
+                        || operation == '^')) {
+                    calculator.calculate(5, 1, operation);
+                }
+            }
+        } catch (RuntimeException e) {
+            throw new IndexOutOfBoundsException();
+        }
+        Assert.fail("Index Out Of Bounds Exception vas expected");
+    }
```

Comment:
```suggestion
    @Test(expected = RuntimeException.class)
  public void unexpectedOperation() {
      for (int i = 0; i < 128; i++) {
        char operation = (char) i;
        if (!(operation == '+'
          || operation == '-'
          || operation == '*'
          || operation == '/'
          || operation == '^')) {
          calculator.calculate(5, 1, operation);
        }
      }
  }
```
ідея спрощення була у цьому
так само можна робити будь-де, коли ти очікуєш, що вилитить ексепшин